### ISSUE TITLE
--tdt poo errors with chrX variants. Inconsistent with plink 1.07

### ISSUE THREAD

**Issue by dantaki** [chrchang/plink-ng#90]
I'm using plink v1.9 to determine parent of origin of singleton variants in WGS. I've noticed that I am missing a lot of chrX variants, and after some digging I believe there is an error in plink v1.9.

For this variant, the mother carries one copy of the ALT allele and transmits it to two sons (one case, one control)

```
$ tabix -h big.chrX.vcf.gz chrX:30242445-30242445 | bcftools query -f'[%SAMPLE\t%GT\n]' -e'GT~"\."' | grep -v "0/0"

Male_Case 1/1
Mom 0/1
Male_Control 1/1
```

I then used plink v1.9 to convert the VCF into a plink bfile and then updated the ids, parents, and genders of the samples. I then use the `--freq counts` command to find singleton variants with success. 

```
$ plink --bfile test.final --freq counts --out test.final

$ more test.final.frq.counts
 CHR  SNP   A1   A2     C1     C2     G0
  23    .    C    T      1    236   5145
```

This makes sense, the VCF was generated using GATK CombineVariants, so there are a lot of missing genotypes because many of the samples were not joint called. Not a big deal, the allele count is correct, one ALT allele in the founder (mother) 

Now I run plink tdt poo to determine the transmission counts

```
$ plink --bfile test.final --tdt poo --pheno plink_files/affected --out test.final
PLINK v1.90b5 64-bit (14 Nov 2017)             www.cog-genomics.org/plink/1.9/
(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to tmp3.log.
Options in effect:
  --bfile test.final
  --out test.final
  --pheno plink_files/affected
  --tdt poo

64453 MB RAM detected; reserving 32226 MB for main workspace.
Allocated 3225 MB successfully, after larger attempt(s) failed.
1 variant loaded from .bim file.
10015 people (5915 males, 4100 females) loaded from .fam.
2703 phenotype values present after --pheno.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 5303 founders and 4712 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.0315527.
1 variant and 10015 people pass filters and QC.
Among remaining phenotypes, 2703 are cases and 0 are controls.  (7312
phenotypes are missing.)
--tdt poo: Report written to test.final.tdt.poo .


$ more test.final.tdt.poo

CHR  SNP  A1:A2      T:U_PAT    CHISQ_PAT        P_PAT      T:U_MAT    CHISQ_MAT        P_MAT        Z_POO        P_POO
  23    .    C:T          0:0           NA           NA          0:0           NA           NA           NA           NA
```

The T:U_MAT should be `1:0` not `0:0` since it was transmitted to an affected son. 

Now in plink v.1.07
```
$ plink-1.07-x86_64/plink --bfile test.final --tdt --poo --pheno plink_files/affected --out test.final.plink1.07 --noweb

@----------------------------------------------------------@
|        PLINK!       |     v1.07      |   10/Aug/2009     |
|----------------------------------------------------------|
|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |
|----------------------------------------------------------|
|  For documentation, citation & bug-report instructions:  |
|        http://pngu.mgh.harvard.edu/purcell/plink/        |
@----------------------------------------------------------@

Skipping web check... [ --noweb ]
Writing this text to log file [ test.final.plink1.07.log ]
Analysis started: Tue Oct  9 17:18:03 2018

Options in effect:
	--bfile test.final
	--tdt
	--poo
	--pheno plink_files/affected
	--out test.final.plink1.07
	--noweb

** For gPLINK compatibility, do not use '.' in --out **
Reading map (extended format) from [ test.final.bim ]
1 markers to be included from [ test.final.bim ]
Reading pedigree information from [ test.final.fam ]
10015 individuals read from [ test.final.fam ]
0 individuals with nonmissing phenotypes
Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)
Missing phenotype value is also -9
0 cases, 0 controls and 10015 missing
5915 males, 4100 females, and 0 of unspecified sex
Reading genotype bitfile from [ test.final.bed ]
Detected that binary PED file is v1.00 SNP-major mode
Reading alternate phenotype from [ plink_files/affected ]
2703 individuals with non-missing alternate phenotype
Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)
Missing phenotype value is also -9
2703 cases, 0 controls and 7312 missing
Before frequency and genotyping pruning, there are 1 SNPs
5303 founders and 4712 non-founders found
Total genotyping rate in remaining individuals is 0.0315527
0 SNPs failed missingness test ( GENO > 1 )
0 SNPs failed frequency test ( MAF < 0 )
After frequency and genotyping pruning, there are 1 SNPs
After filtering, 2703 cases, 0 controls and 7312 missing
After filtering, 5915 males, 4100 females, and 0 of unspecified sex
2653 nuclear families, 0 founder singletons found
4712 non-founders with 2 parents in 2653 nuclear families
0 non-founders without 2 parents in 0 nuclear families
2703 affected offspring trios
0 phenotypically discordant parent pairs found
Converting data to Individual-major format
0 Mendel errors detected in total
Writing TDT parent-of-origin results (asymptotic) to [ test.final.plink1.07.tdt.poo ]

Analysis finished: Tue Oct  9 17:18:04 2018

$ more test.final.plink1.07.tdt.poo

 CHR  SNP  A1:A2      T:U_PAT    CHISQ_PAT        P_PAT      T:U_MAT    CHISQ_MAT        P_MAT        Z_POO        P_POO
  23    .    C:T          0:0           NA           NA          1:0            1       0.3173           NA           NA
```

This is the result I expect. I just wanted to bring this to your attention and sorry for the long issue. I wasn't sure if I was crazy or not.

**Comment by chrchang** [comment]
Can you provide a minimal dataset to reproduce this with?  I just tried creating a tiny .ped and .map matching your description of the relevant family, and did not see what you're describing.

**Comment by dantaki** [comment]
Sure thing, thanks for your help. I attached a tar ball with a VCF of two families, the one I used above and other with REF genotypes in all individuals. There is a README describing the file contents and a script that contains the commands I used. The results I generated with this test case can be found under the `results/` directory. 
[plink_debug.tar.gz](https://github.com/chrchang/plink-ng/files/2466001/plink_debug.tar.gz)
